### ISSUE TITLE
Identation broken after statements with closing parenthesis followed by more code

### ISSUE THREAD

**Issue by kbauer** [spyder-ide/spyder#4074]
## Description of your problem

Indentation is broken, if a line contains a closing parenthesis followed by more code. 

Additionally, comments seem to break the behaviour of `<enter>` auto-indentation as well.

## Reproducible examples

In the following screenshot the "weird indentation" was produced by automatic indentation when pressing "Enter":

![spyder-indent-1](https://cloud.githubusercontent.com/assets/7498468/22425241/7254c3d8-e6fa-11e6-9b72-b27aa9b45d46.png)

## Versions and main components

* Spyder Version: 3.1.2
* Python Version: `3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)]`
* Operating system: Windows 10 Professional, 64bit

## Dependencies

```
jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
numpy >=1.7      :  1.11.1 (OK)
pandas >=0.13.1  :  0.18.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.6.5 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.4.6 (OK)
sympy >=0.7.3    :  1.0 (OK)
```

**Comment by ccordoba12** [comment]
@goanpeca, let's use `3.1.{3, 4, etc}` for really critical things, or things that we didn't finished on time for 3.1 but were almost ready.

All the rest has to go to `3.2`. Else we could end up fixing bugs for new things we introduce in the hotfix branch :-)

**Comment by kbauer** [comment]
Related: Is it possible to disable automatic indentation entirely?

**Comment by rlaverde** [comment]
> Related: Is it possible to disable automatic indentation entirely?

No, It isn't possible

Also, I'm working in the issues that you mentioned, It'll be fixed in v3.2
